crates/iceberg/src/spec/table_metadata.rs

```diff
@@ -514,20 +501,20 @@ impl TryFrom<TableMetadataV1> for TableMetadata {
             partition_specs,
             schemas,
 
-            properties: value.properties,
+            properties: value.properties.unwrap_or_default(),
```

Comment:
This comment is related to next line of `current_snapshot_id`. It seems -1 means empty `current_snapsht_id`, so we should check it when assigning.

Reply:
CC @JanKaul